src/CommentsGridPage.js

```diff
@@ -1,98 +1,36 @@
-        params: {
-          sort: 'new',
-          limit: 30, // not 100 for now because of bad performance of fetching user details individually
-        },
-      });
-
-      return response.data?.data?.children?.map((obj) => obj.data);
-    },
-    {
-      keepPreviousData: true, // so the data doesnt disappear
-      refetchInterval: 15 * 1000,
-    }
-  );
-
-  // fetching authors
-  const authors = useQuery(
-    ['authors', { comments: comments.data }],
-    async ({ queryKey: [, { comments }] }) => {
-      // TODO
-      // better to batch fetch the list but not sure how to do that ATM...
-      const authorsDetails = await Promise.all(
-        comments.map(async (comment) => {
-          const response = await axios.get(`https://www.reddit.com/user/${comment.author}/about/.json`);
-          return response.data.data;
-        })
-      );
+import socketIOClient from 'socket.io-client';
+const ENDPOINT = 'http://127.0.0.1:5000/';
 
-      return authorsDetails;
-    },
-    {
-      enabled: !!comments.data,
-    }
-  );
+export const CommentsGridPage = () => {
+  const [comments, setComments] = useState([]);
 
-  // grouping the comments by author criteria
-  // for now grouping by hardcoded account ages
-  const commentGroups = useMemo(() => {
-    if (authors.data && comments.data) {
-      const groups = groupBy(comments.data, (comment) => {
-        const commentAuthorDetails = authors.data.find((author) => comment.author === author.name);
-        const authorAge = moment(commentAuthorDetails.created_utc * 1000);
-        if (authorAge.isBefore(moment().subtract(6, 'years'))) {
-          return 1;
-        } else if (authorAge.isBefore(moment().subtract(3, 'years'))) {
-          return 2;
-        } else if (authorAge.isBefore(moment().subtract(1, 'years'))) {
-          return 3;
-        } else {
-          return 4;
-        }
+  useEffect(() => {
+    const socket = socketIOClient(ENDPOINT);
+    socket.on('new comment', (data) => {
+      setComments((comments) => {
+        return [data, ...comments];
       });
-
-      return groups;
-    }
-
-    return null;
-  }, [authors, comments]);
+    });
+  }, []);
 
   return (
     <>
       {/* saved in local storage or a cookie */}
       <SortBySelector initialSort={SortEnum.ACCOUNT_AGE} />
 
-      {!!commentGroups && (
-        <Grid container spacing={3}>
-          {[1, 2, 3, 4].map((groupIndex) => {
-            return (
-              <Grid key={groupIndex} item xs={12} sm={6} md={3}>
-                {commentGroups[groupIndex]?.map?.((comment) => {
-                  const author = authors.data.find((author) => comment.author === author.name);
-                  return <Comment key={comment.id} comment={comment} author={author} />;
-                })}
-              </Grid>
-            );
-          })}
+      <Grid container spacing={3}>
+        <Grid key={1} item xs={12} sm={6} md={3}>
+          {comments.map((comment) => (
```

Comment:
good job getting an example up. does removing grouping break the layout?

Reply:
Yes completely breaks the layout

Reply:
ok I can pull your branch into stockticker branch and then add the grouping to your code. will that work for you?